implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/StatsdMeterRegistry.java

```diff
@@ -237,20 +234,30 @@ private void prepareUdpClient(Publisher<String> publisher) {
     }
 
     private void prepareTcpClient(Publisher<String> publisher) {
-        TcpClient.create()
+        AtomicReference<TcpClient> tcpClientReference = new AtomicReference<>();
+        TcpClient tcpClient = TcpClient.create()
                 .host(statsdConfig.host())
                 .port(statsdConfig.port())
                 .handle((in, out) -> out
-                        .options(NettyPipeline.SendOptions::flushOnEach)
                         .sendString(publisher)
                         .neverComplete())
-                .connect()
-                .subscribe(client -> {
-                    this.client.replace(client);
+                .doOnDisconnected(connection -> connectAndSubscribe(tcpClientReference.get()));
```

Comment:
This callback covers the case when the TcpClient *was connected* to a backend but got disconnected - it will try to reconnect.